test/api/positions-deletion.test.js

```diff
@@ -0,0 +1,91 @@
+const { startServer, stopServer } = require('../../lib/server.js');
+const { request } = require('../scripts/helpers');
+const mock = require('../scripts/mock-core-registry');
+const generator = require('../scripts/generator');
+
+describe('Positions deletion', () => {
+    beforeAll(async () => {
+        await startServer();
+    });
+
+    afterAll(async () => {
+        await stopServer();
+    });
+
+    beforeEach(async () => {
+        mock.mockAll();
+    });
+
+    afterEach(async () => {
+        mock.cleanAll();
+        await generator.clearAll();
+    });
+
+    test('should return 403 if no permissions', async () => {
+        mock.mockAll({ mainPermissions: { noPermissions: true } });
+
+        const event = await generator.createEvent({ type: 'agora', applications: [] });
+        const position = await generator.createPosition({}, event);
+
+        const res = await request({
+            uri: '/events/' + event.id + '/positions/' + position.id,
+            method: 'DELETE',
+            headers: { 'X-Auth-Token': 'blablabla' }
+        });
+
+        expect(res.statusCode).toEqual(403);
+        expect(res.body.success).toEqual(false);
+        expect(res.body).toHaveProperty('message');
+        expect(res.body).not.toHaveProperty('data');
+    });
+
+    test('should return 404 if position is not found', async () => {
+        mock.mockAll({ mainPermissions: { noPermissions: true } });
+
+        const event = await generator.createEvent({ type: 'agora', applications: [] });
+
+        const res = await request({
+            uri: '/events/' + event.id + '/positions/1337',
+            method: 'DELETE',
+            headers: { 'X-Auth-Token': 'blablabla' }
+        });
+
+        expect(res.statusCode).toEqual(404);
+        expect(res.body.success).toEqual(false);
+        expect(res.body).toHaveProperty('message');
+        expect(res.body).not.toHaveProperty('data');
+    });
+
+    test('should fail if positions ID is NaN', async () => {
+        const event = await generator.createEvent({ type: 'agora', applications: [] });
+        await generator.createPosition({}, event);
+
+        const res = await request({
+            uri: '/events/' + event.id + '/positions/NaN',
+            method: 'DELETE',
+            headers: { 'X-Auth-Token': 'blablabla' }
+        });
+
+        expect(res.statusCode).toEqual(400);
+        expect(res.body.success).toEqual(false);
+        expect(res.body).toHaveProperty('message');
+        expect(res.body).not.toHaveProperty('data');
+    });
+
+    test('should succeed if everything\'s okay', async () => {
+        const event = await generator.createEvent({ type: 'agora', applications: [] });
+        const position = await generator.generatePosition({}, event);
```

Comment:
*await `generator.createPosition`

Reply:
Ah yes copy-pasting can be difficult at times 😅

Reply:
PS I will fix all this testing mess in the commits, but couldn't get the tests to run locally yet